library/common/http/dispatcher.cc

```diff
@@ -51,16 +52,50 @@ Dispatcher::DirectStream::DirectStream(envoy_stream_t stream_handle,
     : stream_handle_(stream_handle), underlying_stream_(underlying_stream),
       callbacks_(std::move(callbacks)) {}
 
-Dispatcher::Dispatcher(Event::Dispatcher& event_dispatcher,
-                       Upstream::ClusterManager& cluster_manager)
-    : event_dispatcher_(event_dispatcher), cluster_manager_(cluster_manager) {}
+Dispatcher::Dispatcher() {
+  event_dispatcher_.store(nullptr);
+  cluster_manager_.store(nullptr);
+}
+
+void Dispatcher::ready(Event::Dispatcher& event_dispatcher,
+                       Upstream::ClusterManager& cluster_manager) {
+  {
```

Comment:
nit: there is no point in having these extra braces as you hold the lock for the length of the function. Same below.